site/src/components/PortForwardButton/PortForwardButton.tsx

```diff
@@ -0,0 +1,156 @@
+import { CodeExample } from "../CodeExample/CodeExample"
+import { HelpTooltipLink, HelpTooltipLinksGroup, HelpTooltipText } from "../Tooltips/HelpTooltip"
+
+export interface PortForwardButtonProps {
+  host: string
+  username: string
+  workspaceName: string
+  agentName: string
+}
+
+const EnabledView: React.FC<PortForwardButtonProps> = (props) => {
+  const { host, workspaceName, agentName, username } = props
+  const styles = useStyles()
+  const [port, setPort] = useState("3000")
+  const { location } = window
+  const urlExample = `${location.protocol}//${port}--${agentName}--${workspaceName}--${username}.${host}`
+
+  return (
+    <Stack direction="column" spacing={1}>
+      <HelpTooltipText>
+        You can port forward this resource by typing the{" "}
+        <strong>port, workspace name, agent name</strong> and <strong>your username</strong> in the
+        URL like the example below
+      </HelpTooltipText>
+
+      <CodeExample code={urlExample} />
+
+      <HelpTooltipText>Or you can use the following form to open it in a new tab.</HelpTooltipText>
+
+      <Stack direction="row" spacing={1} alignItems="center">
+        <TextField
+          label="Port"
+          type="number"
+          value={port}
+          className={styles.portField}
+          onChange={(e) => {
+            setPort(e.currentTarget.value)
+          }}
+        />
+        <Link
+          underline="none"
+          href={urlExample}
+          target="_blank"
+          rel="noreferrer"
+          className={styles.openUrlButton}
+        >
+          <Button>Open URL</Button>
+        </Link>
+      </Stack>
+
+      <HelpTooltipLinksGroup>
+        <HelpTooltipLink href="https://coder.com/docs/coder-oss/latest/port-forward">
+          Learn more about port forward
+        </HelpTooltipLink>
+      </HelpTooltipLinksGroup>
+    </Stack>
+  )
+}
+
+const DisabledView: React.FC<PortForwardButtonProps> = () => {
+  return (
+    <Stack direction="column" spacing={1}>
+      <HelpTooltipText>
+        <strong>Looks like your deployment does not have port forward enabled.</strong> Check the
+        docs for more info on how to enable it.
+      </HelpTooltipText>
+
+      <HelpTooltipLinksGroup>
+        <HelpTooltipLink href="https://coder.com/docs/coder-oss/latest/networking/port-forwarding">
+          Learn more about port forward
+        </HelpTooltipLink>
+      </HelpTooltipLinksGroup>
+    </Stack>
+  )
+}
+
+export const PortForwardButton: React.FC<PortForwardButtonProps> = (props) => {
+  const { host } = props
+  const anchorRef = useRef<HTMLButtonElement>(null)
+  const [isOpen, setIsOpen] = useState(false)
+  const id = isOpen ? "schedule-popover" : undefined
+  const styles = useStyles()
+
+  const onClose = () => {
+    setIsOpen(false)
+  }
+
+  return (
+    <>
+      <Button
+        startIcon={<OpenInNewOutlined />}
+        size="small"
+        ref={anchorRef}
+        onClick={() => {
+          setIsOpen(true)
+        }}
```

Comment:
nit: `onClose` is in a function but here you use an anonymous function so they're inconsistent.

Reply:
It is because `onClose` is used in two places but I can create a handler function if you think it is better.

Reply:
Ahhh got it. Your call!